### DISCUSSION TITLE
Forgot Password feature

### DISCUSSION THREAD

**Discussion by stevensuting** [apache/superset#18427] [Ideas]
**The Issue**
Currently if a user forgets their password they need to inform the Superset admin from outside the superset environment via email, slack. After which the admin either:
1.  Runs a CLI command
`superset fab reset-password --username name@domain.com  --password the_new_password`
OR
2. Deactivates the user from Superset UI with the need to change all mandatory fields to prevent referential integrity conflicts on the DB and then recreates the user.

**Desired Solution**
Adding a forgot password option/button on the login page that any user can access themselves to change their passwords would make it so much efficient for Admins to manage.

_Note: The only change/update password feature that exist is after the user has successfully logged in._

[a comment by an automated account is not shown]

**Comment by jessicaxyl** [comment]
yes this would be a great feature!

**Comment by junlincc** [comment]
@jessicaxyl @stevensuting 
We would love to provide Design resource if anyone wants to contribute the implementation. please do reachout to me on Slack :)

**Comment by timnick-snow** [comment]
any updated for this feature?

**Comment by xmnlab** [comment]
this would be a very nice feature

**Comment by kamalkeshavani-aiinside** [comment]
I think this should be added in flask app, since Superset is using flask for the login feature. There was a PR https://github.com/dpgaspar/Flask-AppBuilder/pull/1566 for the same, but it is not progressing
